components/sup/src/manager/service/mod.rs

```diff
@@ -935,14 +891,25 @@ impl Service {
             }
         } else {
             self.check_process();
-            // NOTE: if you need reconfiguration and you DON'T have a
-            // reload script, you're going to restart anyway.
-            if self.needs_reload || self.process_down() || self.needs_reconfiguration {
-                self.reload(launcher);
-                if self.needs_reconfiguration {
-                    // NOTE this only runs the hook if it's defined
-                    self.reconfigure()
-                }
+            // Restart the service if:
+            //   1. the run or post-run hooks have changed
+            //   2. the process is down
+            //   3. there is no reconfigure hook and there was a config change
```

Comment:
This code is so legible, I feel like the comment is redundant. Since the risk of the two getting out of sync is relatively high, I'd suggest removing the comment.